solana/pyth2wormhole/program/src/types/mod.rs

```diff
@@ -95,10 +97,43 @@ pub fn batch_serialize(
     // n_attestations
     buf.extend_from_slice(&(collected.len() as u16).to_be_bytes()[..]);
 
-    for a in collected {
-        buf.append(&mut PriceAttestation::serialize(a.borrow()))
+    let mut attestation_size = None; // Will be determined as we serialize attestations
+    let mut serialized_attestations = Vec::with_capacity(collected.len());
+    for (idx, a) in collected.iter().enumerate() {
+        // Learn the current attestation's size
+        let serialized = PriceAttestation::serialize(a.borrow());
+        let a_len = serialized.len();
+
+        // Verify it's the same as the first one we saw for the batch, assign if we're first.
+        if let Some(s) = attestation_size {
+            if s != a_len {
+                return Err(format!(
+                    "attestation {} serializes to {} bytes, {:?} expected",
+                    idx + 1,
+                    a_len,
+                    attestation_size
+                )
+                .into());
+            }
+        } else {
+            attestation_size = Some(serialized.len());
+        }
+
+        serialized_attestations.push(serialized);
+    }
+
+    // Fail if no attestations were serialized
+    let attestation_size =
+        attestation_size.ok_or_else(|| format!("No attestations to serialize"))?;
```

Comment:
imo would be better to set attestation_size = 0 here than to fail. size = 0 shouldn't affect deserialization code (because that code knows the message array is empty).

I wouldn't expect people to be sending around messages with 0 PriceAttestations, but it's generally a good idea to support 0-sized lists. Every time I have made a data structure that doesn't support 0-size for whatever reason, I have regretted it.

Reply:
You're right, I think I've put a good check in a wrong place.